---
title: "pre-processing_CH4trophy_networks"
author: "Rhiannon_Mondav"
date: "1/25/2021"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```


## remove Archaea?
check coverage on SILVA TestPrime : 
96% of bacterial sequences and 95% of archaeal sequences with SILVA, EMBL-ENA, RDP or 98% and 99% respectively against GTDP and LTP for the same SSU rRNA region. 
-decision: don't remove
## remove Eukaryota? Chloroplasts?
Eukaryote coverage is variable between 11% and 17% of available sequences covering the same region. Coverage of chloroplast regions averages over 93% coverage of available sequences. No direct (non-chloroplast) eukaryote sequences were recovered with amplicons. Mitochondrial sequences?
-decision: don't remove chloroplast

https://www.arb-silva.de/search/testprime/
Klindworth, A., Pruesse, E., Schweer, T., Peplies, J., Quast, C., Horn, M. and GlÃ¶ckner, F.O. (2012) Evaluation of general 16S ribosomal RNA gene PCR primers for classical and next-generation sequencing-based diversity studies. Opens external link in new windowNucleic Acids Res. 2013 Jan 1;41(1):e1. doi: 10.1093/nar/gks808

## set up git repository
```{bash}
cd /proj/snic2020-16-196/nobackup/CHtrophy_network_nobackup/
interactive -n 1 -t 02:10:00 -A snic2020-5-529
module load bioinfo-tools git/2.28.0
git init
git clone git@github.com:rmondav/methanotrophy_networks.git
cd methanotrophy_networks/
#add to ignore file
nano .gitignore
# make directories
mkdir original_tables
mkdir scripts
mkdir processed_tables
mkdir network_analyses
mkdir figures
#copy over original mothur tables
scp ../cultALL.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.agc.* ./original_tables/
# remove write access to tables
chmod -w original_tables/cultALL.trim*
#update github
git status
git add .gitignore
git commit -m "update gitignore file"
git push origin main
```


#remove unknown samples and convert to biom format
```{bash}
cd /proj/snic2020-16-196/nobackup/CHtrophy_network_nobackup/
interactive -n 1 -t 02:10:00 -A snic2020-5-529
module load bioinfo-tools Mothur/1.25.1 git/2.28.0
#preprocessing OTU table with mothur#
https://mothur.org/wiki/mothur_manual/

## change name of file while keeping original as backup copy
scp original_tables/cultALL.trim.contigs.good.unique.good.filter.unique.precluster.pick.pick.agc.shared processed_tables/methanotrophy_OTU_table.txt
chmod +w processed_tables/methanotrophy_OTU_table.txt
## remove unknown samples
mothur  
# remove unknown samples
remove.groups(shared=processed_tables/methanotrophy_OTU_table.txt, groups=AnaerobicPLUSC-JamtlaPLUSC-k1-k145-k49-k97)
# convert to biom format
make.biom(shared=processed_tables/methanotrophy_OTU_table.0.03.pick.txt)
quit()
#output methanotrophy_OTU_table.0.03.pick.0.03.biom
```

# separate sample types, remove failed samples and ultra-rare OTUs, then subsample
```{bash}
cd /proj/snic2020-16-196/nobackup/CHtrophy_network_nobackup/methanotrophy_networks
module load bioinfo-tools Qiime/1.9.1 git/2.28.0
## separate membrane filtrate control samples
##remove from main dataset
filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table.0.03.pick.0.03.biom \
-o processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.biom \
--sample_id_fp scripts/sample_list.2umfilterable_controls.txt --negate_sample_id_fp
##only filtrate samples
filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table.0.03.pick.0.03.biom \
-o processed_tables/methanotrophy_OTU_table_filtrate_NTC.biom \
--sample_id_fp scripts/sample_list.2umfilterable_controls.txt 

## make summaries to find failed samples
biom summarize-table \
-i processed_tables/methanotrophy_OTU_table_filtrate_NTC.biom >processed_tables/summary_NTC.txt
biom summarize-table \
-i processed_tables/methanotrophy_OTU_table_filtrate_NTC.biom --qualitative >processed_tables/summary_NTC_qual.txt
biom summarize-table \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.biom >processed_tables/summary_full_woNTC.txt
biom summarize-table \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.biom --qualitative >processed_tables/summary_full_woNTC_qual.txt

## remove samples with less than 500 count as failed samples
filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.biom \
-o processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count.biom \
--min_count 500

## remove ultra-rare = zero count and singleton OTUs
filter_otus_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count.biom \
-o processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count_no_singletons.biom -n 2
#filtrate remove singltns
filter_otus_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_filtrate_NTC.biom \
-o processed_tables/methanotrophy_OTU_table_filtrate_NTC_no_singletons.biom -n 2 

## separate the four 'treatments' for data cleanup
filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count_no_singletons.biom \
-o processed_tables/OTU_table_anO2_lake_cultures.biom \
--sample_id_fp scripts/sample_list.anO2_lake_cultures.txt --negate_sample_id_fp

filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count_no_singletons.biom \
-o processed_tables/OTU_table_anO2_pond_cultures.biom \
--sample_id_fp scripts/sample_list.anO2_pond_cultures.txt --negate_sample_id_fp

filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count_no_singletons.biom \
-o processed_tables/OTU_table_O2_lake_cultures.biom \
--sample_id_fp scripts/sample_list.O2_lake_cultures.txt --negate_sample_id_fp

filter_samples_from_otu_table.py \
-i processed_tables/methanotrophy_OTU_table_wo_uknowns.noNTC.ovr500count_no_singletons.biom \
-o processed_tables/OTU_table_TS_lake.biom \
--sample_id_fp scripts/sample_list.TS_lake_actual.txt --negate_sample_id_fp

# update git
git status
git add ./scripts/ #(etc)
git commit -m "add sample lists"
git push origin main


```{bash}

cd /proj/snic2020-16-196/nobackup/CHtrophy_network_nobackup/methanotrophy_networks/processed_tables
interactive -n 1 -t 02:10:00 -A snic2020-5-529
module load bioinfo-tools Qiime/1.9.1 git/2.28.0

# summarise
biom summarize-table -i OTU_table_anO2_lake_cultures.biom >summary_anO2_lake_cultures.txt #167 S
biom summarize-table -i OTU_table_O2_lake_cultures.biom >summary_O2_lake_cultures.txt #101 s
biom summarize-table -i OTU_table_anO2_pond_cultures.biom >summary_anO2_pond_cultures.txt #96 s
biom summarize-table -i OTU_table_TS_lake.biom >summary_TS_lake.txt #35 s

## subsample 600 counts NTC
single_rarefaction.py \
-i methanotrophy_OTU_table_filtrate_NTC.biom -o OTU_table_filtrate_NTC.600.biom --depth 600
biom summarize-table -i OTU_table_filtrate_NTC.600.biom >summary_NTC.600.txt
biom summarize-table -i OTU_table_filtrate_NTC.600.biom --qualitative >summary_NTC.600.qual.txt
filter_otus_from_otu_table.py \
-i OTU_table_filtrate_NTC.600.biom -o OTU_table_filtrate_NTC.600.no_singletons.biom -n 2
biom summarize-table -i OTU_table_filtrate_NTC.600.no_singletons.biom >summary_NTC.600.ns.txt
biom summarize-table -i OTU_table_filtrate_NTC.600.no_singletons.biom --qualitative >summary_NTC.600.ns.qual.txt

## subsample 2000 counts cultures
single_rarefaction.py \
-i OTU_table_anO2_lake_cultures.biom -o OTU_table_anO2_lake_cultures_2000.biom --depth 2000
biom summarize-table -i OTU_table_anO2_lake_cultures_2000.biom >summary_anO2_lake_cultures_2000.txt #163 samples

single_rarefaction.py \
-i OTU_table_anO2_pond_cultures.biom -o OTU_table_anO2_pond_cultures_2000.biom --depth 2000
biom summarize-table -i OTU_table_anO2_pond_cultures_2000.biom >summary_anO2_pond_cultures_2000.txt #95 samples

single_rarefaction.py \
-i OTU_table_O2_lake_cultures.biom -o OTU_table_O2_lake_cultures_2000.biom --depth 2000
biom summarize-table -i OTU_table_O2_lake_cultures_2000.biom >summary_O2_lake_cultures_2000.txt #101 samples

## subsample at 10 000 count TS lake cultures
single_rarefaction.py \
-i OTU_table_TS_lake.biom -o OTU_table_TS_lake_17000.biom --depth 17000
biom summarize-table -i OTU_table_TS_lake_17000.biom >summary_TS_lake_cultures_17000.txt #35 samples
## subsample same as cultures->useful if we want to ask if subsamples are same as random sampling or result of reduced assemblage culturing. but as we can only ask this of some cultures tehn not particularly useful unless we decide to do a separate MS on only tose culutres
single_rarefaction.py \
-i OTU_table_TS_lake.biom -o OTU_table_TS_lake_2000.biom --depth 2000
biom summarize-table -i OTU_table_TS_lake_2000.biom >summary_TS_lake_cultures_2000.txt

## for network calculations
# try out several options to find something that reduces sparcity to around 50% without removing all the richness/diversity
--min_samples 30% # reduced the cultures too drastically. not good option for cultures as they all had different starting assemblages
--min_samples 10 # slightly better option and based on the idea that 10 samples could give sufficient co-occurance patterns for strong stats
--min_count_fraction 0.1 #retains OTUs with >10% r.a. overall, not good, only dominant OTUs
--min_count_fraction 0.01 #retains OTUs with >1% overal r..a. ok for natural assemblages (lake) and the aerobic cultures but not good option for anO2 cultures
# try combination of >10% of samples  and count (1% one sample + 1x10%) for anO2 cultures and >106 (1% one sample +) for O2 cultures
filter_otus_from_otu_table.py -i OTU_table_anO2_lake_cultures_2000.biom -o OTU_table_anO2_lake_cultures_2000_min17S.biom --min_samples 17
filter_otus_from_otu_table.py -i OTU_table_anO2_lake_cultures_2000_min17S.biom -o OTU_table_anO2_lake_cultures_2000_min17S.ovr36count.biom -n 36
biom summarize-table -i OTU_table_anO2_lake_cultures_2000_min17S.ovr36count.biom >summary_anO2_lake_cultures_2000_min17S.ovr36count.txt
biom summarize-table -i OTU_table_anO2_lake_cultures_2000_min17S.ovr36count.biom --qualitative >summary_anO2_lake_cultures_2000_min17S.ovr36count_qual.txt
#same
filter_otus_from_otu_table.py -i OTU_table_anO2_pond_cultures_2000.biom -o OTU_table_anO2_pond_cultures_2000_min11S.biom --min_samples 11
filter_otus_from_otu_table.py -i OTU_table_anO2_pond_cultures_2000_min11S.biom -o OTU_table_anO2_pond_cultures_2000_min11S.ovr30count.biom -n 30
biom summarize-table -i OTU_table_anO2_pond_cultures_2000_min11S.ovr30count.biom >summary_anO2_pond_cultures_2000_min11S.ovr30count.txt
biom summarize-table -i OTU_table_anO2_pond_cultures_2000_min11S.ovr30count.biom --qualitative >summary_anO2_pond_cultures_2000_min11S.ovr30count_qual.txt
#same ratios for O2 culture
filter_otus_from_otu_table.py -i OTU_table_O2_lake_cultures_2000.biom -o OTU_table_O2_lake_cultures_2000_min11S.biom --min_samples 11
filter_otus_from_otu_table.py -i OTU_table_O2_lake_cultures_2000_min11S.biom -o OTU_table_O2_lake_cultures_2000_min11S.ovr30count.biom -n 30
biom summarize-table -i OTU_table_O2_lake_cultures_2000_min11S.ovr30count.biom >summary_O2_lake_cultures_2000_min11S.ovr30count.txt
biom summarize-table -i OTU_table_O2_lake_cultures_2000_min11S.ovr30count.biom --qualitative >summary_O2_lake_cultures_2000_min11S.ovr30count_qual.txt


## lake samples >50% of samples >= 17.5, 1% in one plus present in 50% of samples = 188
filter_otus_from_otu_table.py -i OTU_table_TS_lake_17000.biom -o OTU_table_TS_lake_17000_min18S.biom --min_samples 18
filter_otus_from_otu_table.py -i OTU_table_TS_lake_17000_min18S.biom -o OTU_table_TS_lake_17000_min18S.ovr189count.biom -n 189
biom summarize-table -i OTU_table_TS_lake_17000_min18S.ovr189count.biom >summary_TS_lake_17000_min18S.ovr189count.txt
biom summarize-table -i OTU_table_TS_lake_17000_min18S.ovr189count.biom --qualitative >summary_TS_lake_17000_min18S.ovr189count_qual.txt

# cultures have sparcity ~70% while lake sample ~10% will try network at this level

###############################
## for community composition ##
#Tried 1% but too extreme, so try 1% in at least one sample
filter_otus_from_otu_table.py -i OTU_table_anO2_lake_cultures_2000.biom -o OTU_table_anO2_lake_cultures_2000.ovr20count.biom -n 20
filter_otus_from_otu_table.py -i OTU_table_anO2_pond_cultures_2000.biom -o OTU_table_anO2_pond_cultures_2000.ovr20count.biom -n 20
filter_otus_from_otu_table.py -i OTU_table_O2_lake_cultures_2000.biom -o OTU_table_O2_lake_cultures_2000.ovr20count.biom -n 20
filter_otus_from_otu_table.py -i OTU_table_TS_lake_17000.biom -o OTU_table_TS_lake_17000.ovr170count.biom -n 170
biom summarize-table -i OTU_table_anO2_lake_cultures_2000.ovr20count.biom >summary_anO2_lake_cultures_2000.ovr20count.txt
biom summarize-table -i OTU_table_anO2_pond_cultures_2000.ovr20count.biom >summary_anO2_pond_cultures_2000.ovr20count.txt
biom summarize-table -i OTU_table_O2_lake_cultures_2000.ovr20count.biom >summary_O2_lake_cultures_2000.ovr20count.txt
biom summarize-table -i OTU_table_TS_lake_17000.ovr170count.biom >summary_TS_lake_17000.ovr170count.txt

#good enough to start with for graphing can change later depending on what story people are interested in, or do taxa summary with original subsampled tables
```


